dot/parachain/gossip-support/gossip_support.go

```diff
@@ -125,14 +309,346 @@ func (gs *GossipSupport) processMessage(msg any) error {
+			logger.Debugf("Low connectivity (due to authority lookup failures) - expected on startup.")
+		}
+	} else {
+		gs.lastFailure = nil
+		gs.failureStart = nil
+	}
+}
+
+func (gs *GossipSupport) issueConnectionRequestToChanged(authorities []parachaintypes.AuthorityDiscoveryID) {
+	_, resolved, _ := gs.resolveAuthorities(authorities)
+
+	changed := make(map[multiaddr.Multiaddr]struct{})
+
+	for authority, newAddresses := range resolved {
+		newPeerIDs := make(map[peer.ID]struct{})
+		for addr := range newAddresses {
+			_, peerID := peer.SplitAddr(addr)
+			newPeerIDs[peerID] = struct{}{}
+		}
+
+		oldAddresses := gs.resolvedAuthorities[authority]
+		if oldAddresses != nil {
+			oldPeerIDs := make(map[peer.ID]struct{})
+			for addr := range oldAddresses {
+				_, peerID := peer.SplitAddr(addr)
+				oldPeerIDs[peerID] = struct{}{}
+			}
+			if !isSuperSet(oldPeerIDs, newPeerIDs) {
+				changed = newAddresses
+			}
+		} else {
+			changed = newAddresses
+		}
+	}
+
+	logger.Debugf("Issuing a connection request to changed validators")
+
+	if len(changed) == 0 {
+		gs.resolvedAuthorities = resolved
+
+		gs.subSystemToOverseer <- networkbridgemessages.AddToResolvedValidators{
+			ValidatorAddrs: changed,
+			PeerSet:        networkbridgemessages.ValidationProtocol,
+		}
+	}
+}
+
+// resolveAuthorities parses the validator address and resolved authorities along with the failure time of parsing
+func (gs *GossipSupport) resolveAuthorities(
+	authorities []parachaintypes.AuthorityDiscoveryID,
+) (map[multiaddr.Multiaddr]struct{}, map[parachaintypes.AuthorityDiscoveryID]map[multiaddr.Multiaddr]struct{}, uint) {
+	validatorAddrs := make(map[multiaddr.Multiaddr]struct{})
+	resolved := make(map[parachaintypes.AuthorityDiscoveryID]map[multiaddr.Multiaddr]struct{}, len(authorities))
+	var failures uint
+
+	for _, authority := range authorities {
+		addrs := gs.authorityDiscovery.GetAddressesByAuthorityID(authority)
+
+		if addrs != nil {
+			validatorAddrs = addrs
+			resolved[authority] = addrs
+		} else {
+			failures++
+			logger.Debugf("Couldn't resolve addresses of authority: %v", authority)
+		}
+	}
+
+	return validatorAddrs, resolved, failures
+}
+
+// isSuperSet returns true if the superset is a superset of subset
+func isSuperSet[K, V comparable](superset, subset map[K]V) bool {
+	for k, v := range subset {
+		if val, ok := superset[k]; !ok || val != v {
+			return false
+		}
+	}
+	return true
+}
+
+// ensureIamAnAuthority return an error if we're not a validator in the given set (do not have keys). Otherwise,
+// returns the index of our keys in authorities.
+func ensureIamAnAuthority(ks keystore.Keystore, authorities []parachaintypes.AuthorityDiscoveryID) (uint, error) {
+	for i, authority := range authorities {
+		publicKey, err := sr25519.NewPublicKey(authority[:])
+		if err != nil {
+			continue
+		}
+		authKey := ks.GetKeypair(publicKey)
+		if authKey != nil {
+			return uint(i), nil
+		}
+	}
+
+	return 0, fmt.Errorf("node is not a validator")
 }
```

Comment:
return boolean instead of error

```suggestion
func ensureIamAnAuthority(ks keystore.Keystore, authorities []parachaintypes.AuthorityDiscoveryID) (uint, bool) {
	for i, authority := range authorities {
		publicKey, err := sr25519.NewPublicKey(authority[:])
		if err != nil {
			continue
		}
		authKey := ks.GetKeypair(publicKey)
		if authKey != nil {
			return uint(i), true
		}
	}

	return 0, false
}
```